packages/connect-node/src/grpc-web-http-transport.ts

```diff
@@ -0,0 +1,421 @@
+            );
+            const conn: StreamingConn<I, O> = {
+              ...req,
+              responseHeader,
+              responseTrailer,
+              closed: false,
+              async send(message: PartialMessage<I>) {
+                if (stream.writableEnded) {
+                  throw new ConnectError(
+                    "cannot send, stream is already closed"
+                  );
+                }
+                const enveloped = encodeEnvelope(
+                  messageFlag,
+                  serialize(normalize(message))
+                );
+                await write(stream, enveloped);
+              },
+              async close(): Promise<void> {
+                if (stream.writableEnded) {
+                  throw new ConnectError(
+                    "cannot close, stream is already closed"
+                  );
+                }
+                this.closed = true;
+                await end(stream);
+              },
+              async read(): Promise<ReadableStreamReadResultLike<O>> {
+                const response = await responsePromise;
+                const responseHeader = nodeHeaderToWebHeader(response.headers);
+                assert(
+                  typeof response.statusCode == "number",
+                  "http1 client response is missing status code"
+                );
+                validateResponse(
+                  useBinaryFormat,
+                  response.statusCode,
+                  responseHeader
+                );
+                try {
+                  const result = await readEnvelope(response);
+                  if (result.done) {
+                    if (messageReceived && !endStreamReceived) {
+                      throw new ConnectError("missing trailers");
+                    }
+                    return {
+                      done: true,
+                      value: undefined,
+                    };
+                  }
+                  if ((result.value.flags & trailerFlag) === trailerFlag) {
+                    endStreamReceived = true;
+                    const trailer = grpcWebTrailerParse(result.value.data);
+                    validateGrpcStatus(trailer);
+                    responseTrailer.resolve(trailer);
+                    return {
+                      done: true,
+                      value: undefined,
+                    };
+                  }
+                  messageReceived = true;
+                  return {
+                    done: false,
+                    value: parse(result.value.data),
+                  };
+                } catch (e) {
+                  throw connectErrorFromNodeReason(e);
+                }
+              },
+            };
+            return Promise.resolve(conn);
+          } catch (e) {
+            throw connectErrorFromNodeReason(e);
+          }
+        }
+      );
+    },
+  };
+}
+
+function makeNodeRequest(options: NodeRequestOptions) {
```

Comment:
@timostamm since this is now being used in the connect and gRPC web transport, should we consider making this reusable or continue to leave it at the transport level?